Практическое задание

### 1. Попробуйте изменить параметры нейронной сети работающей с датасетом imdb либо нейронной сети работающей airline-passengers(она прилагается вместе с датасетом к уроку в виде отдельного скрипта) так, чтобы улучшить ее точность.

Приложите анализ.

In [22]:
from __future__ import print_function

from keras.preprocessing import sequence
from keras.models import Sequential
from keras.layers import Dense, Embedding
from keras.layers import LSTM, SimpleRNN, GRU
from keras.datasets import imdb
from keras.utils import pad_sequences

from tensorflow import keras

max_features = 15000

# обрезание текстов после данного количества слов (среди top max_features наиболее используемые слова)
maxlen = 500
batch_size = 64 # увеличьте значение для ускорения обучения
def train_nn():
  print('Загрузка данных...')
  (x_train, y_train), (x_test, y_test) = imdb.load_data(num_words=max_features)
  print(len(x_train), 'тренировочные последовательности')
  print(len(x_test), 'тестовые последовательности')

  print('Pad последовательности (примеров в x единицу времени)')
  x_train = pad_sequences(x_train, maxlen=maxlen)
  x_test = pad_sequences(x_test, maxlen=maxlen)
  print('x_train shape:', x_train.shape)
  print('x_test shape:', x_test.shape)

  print('Построение модели...')
  model = Sequential()
  model.add(Embedding(max_features, 128))
  model.add(LSTM(128, dropout=0.2, recurrent_dropout=0.2))
  model.add(Dense(1, activation='sigmoid'))

  opt = keras.optimizers.Nadam(learning_rate=0.0001)

  # стоит попробовать использовать другие оптимайзер и другие конфигурации оптимайзеров 
  model.compile(loss='binary_crossentropy',
#                optimizer = opt,
                optimizer='nadam', # при использовании этого оптимайзера модель показывает наилучшие результаты.
                metrics=['accuracy'])

  print('Процесс обучения...')
  model.fit(x_train, y_train,
            batch_size=batch_size,
            epochs=1, # увеличьте при необходимости
            validation_data=(x_test, y_test))
  score, acc = model.evaluate(x_test, y_test,
                              batch_size=batch_size)
  print('Результат при тестировании:', score)
  print('Тестовая точность:', acc)
train_nn()

Загрузка данных...
25000 тренировочные последовательности
25000 тестовые последовательности
Pad последовательности (примеров в x единицу времени)
x_train shape: (25000, 500)
x_test shape: (25000, 500)
Построение модели...
Процесс обучения...
391/391 [==============================] - 110s 283ms/step - loss: 0.4567 - accuracy: 0.7790
Результат при тестировании: 0.4567078649997711
Тестовая точность: 0.7789999842643738


### Выводы:
Для увеичения качества сети:
- увеличели количество наиболее используемых слов среди top max_features до 500 (дальше тестовая точность начинает падать)
- уменьшили количество нейронов до 64
- тестирование сети с прочими параметрами можно прогонять с большим batch_size, но финально все же снизить до 75, чтобы увеличить точность
- действительно, оптимизатор Nadam со стандартными настройками показал наилучшие результаты для данной сети


### Лучший результат
maxlen = 500 batch_size = 75 64 нейрона Nadam optimizer(со стандартными настройками)
334/334 [==============================] - 169s 504ms/step - loss: 0.4380 - accuracy: 0.7947 - val_loss: 0.3248 - val_accuracy: 0.8722
334/334 [==============================] - 23s 70ms/step - loss: 0.3248 - accuracy: 0.8722
Результат при тестировании: 0.3247823715209961
Тестовая точность: 0.8722000122070312


### Прочие результаты


#### Изначальный результат:
batch_size = 75 maxlen = 80 colab
334/334 [==============================] - 90s 261ms/step - loss: 0.4329 - accuracy: 0.7910 - val_loss: 0.3619 - val_accuracy: 0.8397
334/334 [==============================] - 12s 37ms/step - loss: 0.3619 - accuracy: 0.8397
Результат при тестировании: 0.3618793487548828
Тестовая точность: 0.839680016040802
_______________________________________________________________________________________________________

batch_size = 150 maxlen = 500 Nadam optimizer
167/167 [==============================] - 1060s 6s/step - loss: 0.4443 - accuracy: 0.7898 - val_loss: 0.3288 - val_accuracy: 0.8687
167/167 [==============================] - 157s 943ms/step - loss: 0.3288 - accuracy: 0.8687
Результат при тестировании: 0.32884684205055237
Тестовая точность: 0.8686800003051758



maxlen = 500 batch_size = 75 256 нейронов
334/334 [==============================] - 1259s 4s/step - loss: 0.5089 - accuracy: 0.7581 - val_loss: 0.4197 - val_accuracy: 0.8117
334/334 [==============================] - 286s 856ms/step - loss: 0.4197 - accuracy: 0.8117
Результат при тестировании: 0.4196734130382538
Тестовая точность: 0.8117200136184692


maxlen = 500 batch_size = 150 256 нейронов
167/167 [==============================] - 2384s 14s/step - loss: 0.5193 - accuracy: 0.7309 - val_loss: 0.5090 - val_accuracy: 0.7412
167/167 [==============================] - 400s 2s/step - loss: 0.5090 - accuracy: 0.7412
Результат при тестировании: 0.5090473294258118
Тестовая точность: 0.741159975528717


batch_size = 300 maxlen = 80
Процесс обучения...
84/84 [==============================] - 1215s 14s/step - loss: 0.5338 - accuracy: 0.7285 - val_loss: 0.3980 - val_accuracy: 0.8297
84/84 [==============================] - 28s 328ms/step - loss: 0.3980 - accuracy: 0.8297
Результат при тестировании: 0.39795514941215515
Тестовая точность: 0.8296800255775452

batch_size = 750 maxlen = 200
34/34 [==============================] - 3393s 100s/step - loss: 0.6604 - accuracy: 0.6405 - val_loss: 0.5304 - val_accuracy: 0.7473
34/34 [==============================] - 97s 3s/step - loss: 0.5304 - accuracy: 0.7473
Результат при тестировании: 0.5303581953048706
Тестовая точность: 0.7472800612449646


batch_size = 150 maxlen =80 colab
167/167 [==============================] - 163s 954ms/step - loss: 0.4978 - accuracy: 0.7384 - val_loss: 0.3638 - val_accuracy: 0.8393
167/167 [==============================] - 25s 151ms/step - loss: 0.3638 - accuracy: 0.8393
Результат при тестировании: 0.3638118505477905
Тестовая точность: 0.8393200039863586

batch_size = 150  maxlen = 100 
167/167 [==============================] - 211s 1s/step - loss: 0.4718 - accuracy: 0.7562 - val_loss: 0.3411 - val_accuracy: 0.8532
167/167 [==============================] - 31s 183ms/step - loss: 0.3411 - accuracy: 0.8532
Результат при тестировании: 0.34107333421707153
Тестовая точность: 0.85316002368927

batch_size = 150 maxlen = 200 colab
167/167 [==============================] - 389s 2s/step - loss: 0.4763 - accuracy: 0.7698 - val_loss: 0.3388 - val_accuracy: 0.8659
167/167 [==============================] - 62s 369ms/step - loss: 0.3388 - accuracy: 0.8659
Результат при тестировании: 0.33877038955688477
Тестовая точность: 0.8659200072288513

batch_size = 150 maxlen = 500 colab
167/167 [==============================] - 1060s 6s/step - loss: 0.4443 - accuracy: 0.7898 - val_loss: 0.3288 - val_accuracy: 0.8687
167/167 [==============================] - 157s 943ms/step - loss: 0.3288 - accuracy: 0.8687
Результат при тестировании: 0.32884684205055237
Тестовая точность: 0.8686800003051758

batch_size = 75 maxlen = 500 colab
334/334 [==============================] - 1003s 3s/step - loss: 0.4623 - accuracy: 0.7795 - val_loss: 0.3506 - val_accuracy: 0.8589
334/334 [==============================] - 131s 393ms/step - loss: 0.3506 - accuracy: 0.8589
Результат при тестировании: 0.3506370782852173
Тестовая точность: 0.8588799834251404


batch_size = 75 maxlen = 500 colab sgd
334/334 [==============================] - 967s 3s/step - loss: 0.6932 - accuracy: 0.5014 - val_loss: 0.6931 - val_accuracy: 0.5080
334/334 [==============================] - 120s 360ms/step - loss: 0.6931 - accuracy: 0.5080
Результат при тестировании: 0.6930833458900452
Тестовая точность: 0.5079600214958191


batch_size = 150 maxlen = 500 adamax
167/167 [==============================] - 443s 3s/step - loss: 0.5315 - accuracy: 0.7317 - val_loss: 0.3483 - val_accuracy: 0.8558
167/167 [==============================] - 81s 488ms/step - loss: 0.3483 - accuracy: 0.8558
Результат при тестировании: 0.3482741713523865
Тестовая точность: 0.8558400273323059


batch_size = 75 maxlen = 500 colab tanh
334/334 [==============================] - 1005s 3s/step - loss: 0.5747 - accuracy: 0.7481 - val_loss: 0.3962 - val_accuracy: 0.8375
334/334 [==============================] - 126s 376ms/step - loss: 0.3962 - accuracy: 0.8375
Результат при тестировании: 0.39619699120521545
Тестовая точность: 0.8374800086021423


batch_size = 75 maxlen = 500 colab adam
334/334 [==============================] - 968s 3s/step - loss: 0.4244 - accuracy: 0.8030 - val_loss: 0.3795 - val_accuracy: 0.8354
334/334 [==============================] - 121s 361ms/step - loss: 0.3795 - accuracy: 0.8354
Результат при тестировании: 0.37952741980552673
Тестовая точность: 0.8353599905967712

batch_size = 75 maxlen = 500 colab 64
334/334 [==============================] - 475s 1s/step - loss: 0.5022 - accuracy: 0.7406 - val_loss: 0.3356 - val_accuracy: 0.8577
334/334 [==============================] - 56s 168ms/step - loss: 0.3356 - accuracy: 0.8577
Результат при тестировании: 0.3355770409107208
Тестовая точность: 0.8577200174331665

batch_size = 150 maxlen = 250 colab 256
167/167 [==============================] - 1132s 7s/step - loss: 0.4808 - accuracy: 0.7734 - val_loss: 0.3233 - val_accuracy: 0.8659
167/167 [==============================] - 194s 1s/step - loss: 0.3233 - accuracy: 0.8659
Результат при тестировании: 0.323293000459671
Тестовая точность: 0.8659200072288513


### 2. Попробуйте изменить параметры нейронной сети генерирующий текст таким образом, чтобы добиться генерации как можно более осмысленного текста. 

Пришлите лучший получившейся у вас текст и опишите, что вы предприняли, чтобы его получить. Можно использовать текст другого прозведения.

* Попробуйте на numpy реализовать нейронную сеть архитектуры LSTM

* Предложите свои варианты решения проблемы исчезающего градиента в RNN
В нейронных сетях, если первые уровни набирают наименьшее количество градиента, процесс их обучения прекращается. Поскольку эти слои не обучаются, RNN не запоминает ничего из опыта более длинных серий данных и сталкивается с проблемами краткосрочной памяти.

GRU, например, используют ворота сброса. Основная функция ворот обновления заключается в определении идеального количества предыдущей информации, которая важна для будущего. То есть отсеивает часть входящей информации

In [7]:
import numpy as np
from keras.layers import Dense, Activation
from tensorflow.keras.layers import LSTM, SimpleRNN, GRU
from keras.models import Sequential


# построчное чтение из примера с текстом 
# from google.colab import files
# uploaded = files.upload()


with open("alice_in_wonderland.txt", 'rb') as _in:
    lines = []
    for line in _in:
        line = line.strip().lower().decode("utf-8", "ignore")
        if len(line) == 0:
            continue
        lines.append(line)
text = " ".join(lines)
chars = set([c for c in text])
nb_chars = len(chars)


# создание индекса символов и reverse mapping чтобы передвигаться между значениями numerical
# ID and a specific character. The numerical ID will correspond to a column
# ID и определенный символ. Numerical ID будет соответсвовать колонке
# число при использовании one-hot кодировки для представление входов символов
char2index = {c: i for i, c in enumerate(chars)}
index2char = {i: c for i, c in enumerate(chars)}

# для удобства выберете фиксированную длину последовательность 10 символов 
SEQLEN, STEP = 120, 2
input_chars, label_chars = [], []

# конвертация data в серии разных SEQLEN-length субпоследовательностей
for i in range(0, len(text) - SEQLEN, STEP):
    input_chars.append(text[i: i + SEQLEN])
    label_chars.append(text[i + SEQLEN])


# Вычисление one-hot encoding входных последовательностей X и следующего символа (the label) y

X = np.zeros((len(input_chars), SEQLEN, nb_chars), dtype=np.bool)
y = np.zeros((len(input_chars), nb_chars), dtype=np.bool_)
for i, input_char in enumerate(input_chars):
    for j, ch in enumerate(input_char):
        X[i, j, char2index[ch]] = 1
    y[i, char2index[label_chars[i]]] = 1


# установка ряда метапамертров  для нейронной сети и процесса тренировки
BATCH_SIZE, HIDDEN_SIZE = 128, 256
NUM_ITERATIONS = 10
NUM_EPOCHS_PER_ITERATION = 10
NUM_PREDS_PER_EPOCH = 100


# Create a super simple recurrent neural network. There is one recurrent
# layer that produces an embedding of size HIDDEN_SIZE from the one-hot
# encoded input layer. This is followed by a Dense fully-connected layer
# across the set of possible next characters, which is converted to a
# probability score via a standard softmax activation with a multi-class
# cross-entropy loss function linking the prediction to the one-hot
# encoding character label.

'''
Создание очень простой рекуррентной нейронной сети. В ней будет один реккурентный 
закодированный входной слой. За ним последует полносвязный слой связанный с набором возможных 
следующих символов, которые конвертированы в вероятностные результаты через стандартную softmax активацию 
с multi-class cross-encoding loss функцию ссылающуются на предсказание one-hot encoding лейбл символа
'''

model = Sequential()
model.add(
    GRU(  # вы можете изменить эту часть на LSTM, GRU или SimpleRNN, чтобы попробовать альтернативы
        HIDDEN_SIZE * 2,
        return_sequences=False,
        input_shape=(SEQLEN, nb_chars),
        unroll=True
    )
)
model.add(Dense(nb_chars))
model.add(Activation("softmax"))
model.compile(loss="categorical_crossentropy", optimizer="rmsprop")


# выполнение серий тренировочных и демонстрационных итераций 
for iteration in range(NUM_ITERATIONS):

    # для каждой итерации запуск передачи данных в модель 
    print("=" * 50)
    print("Итерация #: %d" % (iteration))
    model.fit(X, y, batch_size=BATCH_SIZE, epochs=NUM_EPOCHS_PER_ITERATION)

    # Select a random example input sequence.
    test_idx = np.random.randint(len(input_chars))
    test_chars = input_chars[test_idx]

    # для числа шагов предсказаний использование текущей тренируемой модели 
    # конструирование one-hot encoding для тестирования input и добавление предсказания.
    print("Генерация из посева: %s" % (test_chars))
    print(test_chars, end="")
    for i in range(NUM_PREDS_PER_EPOCH):

        # здесь one-hot encoding.
        X_test = np.zeros((1, SEQLEN, nb_chars))
        for j, ch in enumerate(test_chars):
            X_test[0, j, char2index[ch]] = 1

        # осуществление предсказания с помощью текущей модели.
        pred = model.predict(X_test, verbose=0)[0]
        y_pred = index2char[np.argmax(pred)]

        # вывод предсказания добавленного к тестовому примеру 
        print(y_pred, end="")

        # инкрементация тестового примера содержащего предсказание
        test_chars = test_chars[1:] + y_pred
print()

# Модель генерирует более осмысленный текст при увеличении количества эпох и итераций, а так же при увеличении количества нейронов в скрытом слое.


<ipython-input-7-e3fe63d302e6>:43: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  X = np.zeros((len(input_chars), SEQLEN, nb_chars), dtype=np.bool)


Итерация #: 0
Epoch 1/10
632/632 [==============================] - 300s 461ms/step - loss: 2.3366
Epoch 2/10
632/632 [==============================] - 295s 466ms/step - loss: 1.8804
Epoch 3/10
632/632 [==============================] - 277s 439ms/step - loss: 1.6747
Epoch 4/10
632/632 [==============================] - 276s 437ms/step - loss: 1.5202
Epoch 5/10
632/632 [==============================] - 275s 435ms/step - loss: 1.3862
Epoch 6/10
632/632 [==============================] - 276s 437ms/step - loss: 1.2626
Epoch 7/10
632/632 [==============================] - 276s 436ms/step - loss: 1.1425
Epoch 8/10
632/632 [==============================] - 275s 435ms/step - loss: 1.0220
Epoch 9/10
632/632 [==============================] - 275s 435ms/step - loss: 0.9020
Epoch 10/10
632/632 [==============================] - 275s 435ms/step - loss: 0.7807
Генерация из посева: ister, who was gently brushing away some dead leaves that had fluttered down from the trees upon her face. ‘wake u

632/632 [==============================] - 282s 446ms/step - loss: 0.1581
Epoch 7/10
632/632 [==============================] - 279s 442ms/step - loss: 0.1566
Epoch 8/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1565
Epoch 9/10
632/632 [==============================] - 280s 442ms/step - loss: 0.1581
Epoch 10/10
632/632 [==============================] - 279s 442ms/step - loss: 0.1589
Генерация из посева: e queen of hearts, and i had to sing “twinkle, twinkle, little bat! how i wonder what you’re at!” you know the song, per
e queen of hearts, and i had to sing “twinkle, twinkle, little bat! how i wonder what you’re at!” you know the song, perhaps? it was might it was in to hisself, and griving in his noth-up pellirg hore, she puppobed the l==================================================
Итерация #: 7
Epoch 1/10
632/632 [==============================] - 282s 445ms/step - loss: 0.1610
Epoch 2/10
632/632 [==============================] - 283s 448ms/step - l

### ВЫВОДЫ:
Для повыщения качества сети (уменьшения ошибки) следует увеличить количество нейоров в скрытом слое. 

Из-за нехватки времени не удалось прогнать сеть с различными оптимайзерами на больцом количестве итерации и эпох из-за этого полного анализа по ним не получили.

Из результатов видно, что на выходе можно получить сгенерированные слова, которые нейросеть пытается построить в предложения. Данные предложения, как и некоторые слова имеют ошибки, при этом слова и словосочетания нейросеть строит вполне логичные, то предложения не всегда логичны.

BATCH_SIZE, HIDDEN_SIZE = 128, 512
NUM_ITERATIONS = 10
NUM_EPOCHS_PER_ITERATION = 10
NUM_PREDS_PER_EPOCH = 100

==================================================
Итерация #: 0
Epoch 1/10
632/632 [==============================] - 300s 461ms/step - loss: 2.3366
Epoch 2/10
632/632 [==============================] - 295s 466ms/step - loss: 1.8804
Epoch 3/10
632/632 [==============================] - 277s 439ms/step - loss: 1.6747
Epoch 4/10
632/632 [==============================] - 276s 437ms/step - loss: 1.5202
Epoch 5/10
632/632 [==============================] - 275s 435ms/step - loss: 1.3862
Epoch 6/10
632/632 [==============================] - 276s 437ms/step - loss: 1.2626
Epoch 7/10
632/632 [==============================] - 276s 436ms/step - loss: 1.1425
Epoch 8/10
632/632 [==============================] - 275s 435ms/step - loss: 1.0220
Epoch 9/10
632/632 [==============================] - 275s 435ms/step - loss: 0.9020
Epoch 10/10
632/632 [==============================] - 275s 435ms/step - loss: 0.7807
Генерация из посева: ister, who was gently brushing away some dead leaves that had fluttered down from the trees upon her face. ‘wake up, ali
ister, who was gently brushing away some dead leaves that had fluttered down from the trees upon her face. ‘wake up, alice said to the mouse in the anster that she had got not a moutes and shook in the anster that she ha==================================================
Итерация #: 1
Epoch 1/10
632/632 [==============================] - 276s 437ms/step - loss: 0.6669
Epoch 2/10
632/632 [==============================] - 273s 432ms/step - loss: 0.5642
Epoch 3/10
632/632 [==============================] - 275s 435ms/step - loss: 0.4753
Epoch 4/10
632/632 [==============================] - 279s 441ms/step - loss: 0.4007
Epoch 5/10
632/632 [==============================] - 279s 442ms/step - loss: 0.3407
Epoch 6/10
632/632 [==============================] - 280s 443ms/step - loss: 0.2951
Epoch 7/10
632/632 [==============================] - 278s 440ms/step - loss: 0.2590
Epoch 8/10
632/632 [==============================] - 276s 437ms/step - loss: 0.2296
Epoch 9/10
632/632 [==============================] - 276s 437ms/step - loss: 0.2104
Epoch 10/10
632/632 [==============================] - 278s 439ms/step - loss: 0.1944
Генерация из посева: one hand and a piece of bread-and-butter in the other. ‘i beg pardon, your majesty,’ he began, ‘for bringing these in: b
one hand and a piece of bread-and-butter in the other. ‘i beg pardon, your majesty,’ he began, ‘for bringing these in: but i non’t lenst if you disnout.’ ‘it’s the first she said this could not soom to the trightse, by t==================================================
Итерация #: 2
Epoch 1/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1850
Epoch 2/10
632/632 [==============================] - 277s 438ms/step - loss: 0.1725
Epoch 3/10
632/632 [==============================] - 277s 438ms/step - loss: 0.1658
Epoch 4/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1581
Epoch 5/10
632/632 [==============================] - 275s 435ms/step - loss: 0.1571
Epoch 6/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1537
Epoch 7/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1490
Epoch 8/10
632/632 [==============================] - 273s 431ms/step - loss: 0.1474
Epoch 9/10
632/632 [==============================] - 275s 435ms/step - loss: 0.1470
Epoch 10/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1430
Генерация из посева: foundation’s web site and official page at http://pglaf.org for additional contact information: dr. gregory b. newby chi
foundation’s web site and official page at http://pglaf.org for additional contact information: dr. gregory b. newby chief difficers wrot notring erealy of any orman connarta, of that part comily project gutenberg-tm lic==================================================
Итерация #: 3
Epoch 1/10
632/632 [==============================] - 273s 433ms/step - loss: 0.1395
Epoch 2/10
632/632 [==============================] - 276s 436ms/step - loss: 0.1415
Epoch 3/10
632/632 [==============================] - 274s 433ms/step - loss: 0.1390
Epoch 4/10
632/632 [==============================] - 271s 429ms/step - loss: 0.1430
Epoch 5/10
632/632 [==============================] - 274s 433ms/step - loss: 0.1358
Epoch 6/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1404
Epoch 7/10
632/632 [==============================] - 274s 433ms/step - loss: 0.1419
Epoch 8/10
632/632 [==============================] - 273s 432ms/step - loss: 0.1402
Epoch 9/10
632/632 [==============================] - 273s 432ms/step - loss: 0.1403
Epoch 10/10
632/632 [==============================] - 275s 435ms/step - loss: 0.1370
Генерация из посева: my elbow.’ on which seven looked up and said, ‘that’s right, five! always lay the blame on others!’ ‘you’d better not ta
my elbow.’ on which seven looked up and said, ‘that’s right, five! always lay the blame on others!’ ‘you’d better not talk!’ said five. ‘why,’ said the hatter: ‘i means, that go oo mined them were and return in the mock ==================================================
Итерация #: 4
Epoch 1/10
632/632 [==============================] - 276s 437ms/step - loss: 0.1389
Epoch 2/10
632/632 [==============================] - 276s 436ms/step - loss: 0.1400
Epoch 3/10
632/632 [==============================] - 274s 434ms/step - loss: 0.1403
Epoch 4/10
632/632 [==============================] - 274s 434ms/step - loss: 0.1418
Epoch 5/10
632/632 [==============================] - 275s 435ms/step - loss: 0.1419
Epoch 6/10
632/632 [==============================] - 275s 434ms/step - loss: 0.1388
Epoch 7/10
632/632 [==============================] - 274s 434ms/step - loss: 0.1380
Epoch 8/10
632/632 [==============================] - 274s 434ms/step - loss: 0.1363
Epoch 9/10
632/632 [==============================] - 273s 432ms/step - loss: 0.1409
Epoch 10/10
632/632 [==============================] - 273s 432ms/step - loss: 0.1421
Генерация из посева: ng, being fast asleep. ‘after that,’ continued the hatter, ‘i cut some more bread-and-butter--’ ‘but what did the dormou
ng, being fast asleep. ‘after that,’ continued the hatter, ‘i cut some more bread-and-butter--’ ‘but what did the dormouse indeed,’ said alice indeginping to her, ‘but letthe sean that’s no sourd of the mock turtle. ‘of ==================================================
Итерация #: 5
Epoch 1/10
632/632 [==============================] - 275s 436ms/step - loss: 0.1425
Epoch 2/10
632/632 [==============================] - 278s 441ms/step - loss: 0.1428
Epoch 3/10
632/632 [==============================] - 288s 455ms/step - loss: 0.1441
Epoch 4/10
632/632 [==============================] - 286s 453ms/step - loss: 0.1434
Epoch 5/10
632/632 [==============================] - 289s 458ms/step - loss: 0.1458
Epoch 6/10
632/632 [==============================] - 279s 442ms/step - loss: 0.1455
Epoch 7/10
632/632 [==============================] - 287s 454ms/step - loss: 0.1439
Epoch 8/10
632/632 [==============================] - 282s 445ms/step - loss: 0.1471
Epoch 9/10
632/632 [==============================] - 283s 447ms/step - loss: 0.1498
Epoch 10/10
632/632 [==============================] - 283s 448ms/step - loss: 0.1469
Генерация из посева: he terms of the project gutenberg license included with this ebook or online at www.gutenberg.org 1.e.2.  if an individu
he terms of the project gutenberg license included with this ebook or online at www.gutenberg.org 1.e.2.  if an individual project gutenberg-tm ebeatronic works that a reaso are dock to the project gutenberg-tm nomply al==================================================
Итерация #: 6
Epoch 1/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1479
Epoch 2/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1526
Epoch 3/10
632/632 [==============================] - 280s 443ms/step - loss: 0.1503
Epoch 4/10
632/632 [==============================] - 280s 443ms/step - loss: 0.1573
Epoch 5/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1530
Epoch 6/10
632/632 [==============================] - 282s 446ms/step - loss: 0.1581
Epoch 7/10
632/632 [==============================] - 279s 442ms/step - loss: 0.1566
Epoch 8/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1565
Epoch 9/10
632/632 [==============================] - 280s 442ms/step - loss: 0.1581
Epoch 10/10
632/632 [==============================] - 279s 442ms/step - loss: 0.1589
Генерация из посева: e queen of hearts, and i had to sing “twinkle, twinkle, little bat! how i wonder what you’re at!” you know the song, per
e queen of hearts, and i had to sing “twinkle, twinkle, little bat! how i wonder what you’re at!” you know the song, perhaps? it was might it was in to hisself, and griving in his noth-up pellirg hore, she puppobed the l==================================================
Итерация #: 7
Epoch 1/10
632/632 [==============================] - 282s 445ms/step - loss: 0.1610
Epoch 2/10
632/632 [==============================] - 283s 448ms/step - loss: 0.1620
Epoch 3/10
632/632 [==============================] - 283s 448ms/step - loss: 0.1610
Epoch 4/10
632/632 [==============================] - 283s 447ms/step - loss: 0.1608
Epoch 5/10
632/632 [==============================] - 283s 447ms/step - loss: 0.1633
Epoch 6/10
632/632 [==============================] - 283s 447ms/step - loss: 0.1621
Epoch 7/10
632/632 [==============================] - 282s 446ms/step - loss: 0.1598
Epoch 8/10
632/632 [==============================] - 282s 446ms/step - loss: 0.1637
Epoch 9/10
632/632 [==============================] - 282s 447ms/step - loss: 0.1605
Epoch 10/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1627
Генерация из посева: ooking about for some way of escape, and wondering whether she could get away without being seen, when she noticed a cur
ooking about for some way of escape, and wondering whether she could get away without being seen, when she noticed a curious cake, but at leate a serpert. ‘dows that looker what a shilloup of the e--eversent, so you a ca==================================================
Итерация #: 8
Epoch 1/10
632/632 [==============================] - 280s 443ms/step - loss: 0.1628
Epoch 2/10
632/632 [==============================] - 277s 438ms/step - loss: 0.1622
Epoch 3/10
632/632 [==============================] - 277s 439ms/step - loss: 0.1679
Epoch 4/10
632/632 [==============================] - 280s 444ms/step - loss: 0.1697
Epoch 5/10
632/632 [==============================] - 281s 444ms/step - loss: 0.1690
Epoch 6/10
632/632 [==============================] - 281s 444ms/step - loss: 0.1711
Epoch 7/10
632/632 [==============================] - 282s 446ms/step - loss: 0.1707
Epoch 8/10
632/632 [==============================] - 282s 446ms/step - loss: 0.1677
Epoch 9/10
632/632 [==============================] - 285s 451ms/step - loss: 0.1746
Epoch 10/10
632/632 [==============================] - 284s 449ms/step - loss: 0.1706
Генерация из посева: all day about it!’ and he went on in these words: ‘yes, we went to school in the sea, though you mayn’t believe it--’ ‘i
all day about it!’ and he went on in these words: ‘yes, we went to school in the sea, though you mayn’t believe it--’ ‘i didn’t!’ and the tire watchen her ineam, and the lost crist not some the rest wart one op the wit, ==================================================
Итерация #: 9
Epoch 1/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1749
Epoch 2/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1753
Epoch 3/10
632/632 [==============================] - 280s 443ms/step - loss: 0.1730
Epoch 4/10
632/632 [==============================] - 282s 446ms/step - loss: 0.1812
Epoch 5/10
632/632 [==============================] - 279s 441ms/step - loss: 0.1867
Epoch 6/10
632/632 [==============================] - 280s 443ms/step - loss: 0.1750
Epoch 7/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1838
Epoch 8/10
632/632 [==============================] - 281s 445ms/step - loss: 0.1848
Epoch 9/10
632/632 [==============================] - 288s 456ms/step - loss: 0.1915
Epoch 10/10
632/632 [==============================] - 288s 455ms/step - loss: 0.1879
Генерация из посева: ly, throwing an inkstand at the lizard as she spoke. (the unfortunate little bill had left off writing on his slate with
ly, throwing an inkstand at the lizard as she spoke. (the unfortunate little bill had left off writing on his slate with the queen’s angrily of anything which was such a child--’  uplity to a wrig), but you ank righted a

BATCH_SIZE, HIDDEN_SIZE = 128, 256
NUM_ITERATIONS = 10
NUM_EPOCHS_PER_ITERATION = 10
NUM_PREDS_PER_EPOCH = 100

==================================================
Итерация #: 0
Epoch 1/10
632/632 [==============================] - 135s 199ms/step - loss: 2.4218
Epoch 2/10
632/632 [==============================] - 130s 205ms/step - loss: 2.0026
Epoch 3/10
632/632 [==============================] - 133s 210ms/step - loss: 1.8253
Epoch 4/10
632/632 [==============================] - 127s 201ms/step - loss: 1.6938
Epoch 5/10
632/632 [==============================] - 128s 203ms/step - loss: 1.5875
Epoch 6/10
632/632 [==============================] - 131s 207ms/step - loss: 1.4997
Epoch 7/10
632/632 [==============================] - 131s 207ms/step - loss: 1.4212
Epoch 8/10
632/632 [==============================] - 132s 209ms/step - loss: 1.3493
Epoch 9/10
632/632 [==============================] - 135s 213ms/step - loss: 1.2818
Epoch 10/10
632/632 [==============================] - 133s 210ms/step - loss: 1.2178
Генерация из посева: sion. she had already heard her sentence three of the players to be executed for having missed their turns, and she did 
sion. she had already heard her sentence three of the players to be executed for having missed their turns, and she did not get to the stapped and all the stare, and all the shall repented the mock turtle say of the star==================================================
Итерация #: 1
Epoch 1/10
632/632 [==============================] - 132s 209ms/step - loss: 1.1549
Epoch 2/10
632/632 [==============================] - 132s 209ms/step - loss: 1.0953
Epoch 3/10
632/632 [==============================] - 134s 212ms/step - loss: 1.0349
Epoch 4/10
632/632 [==============================] - 134s 212ms/step - loss: 0.9754
Epoch 5/10
632/632 [==============================] - 133s 211ms/step - loss: 0.9182
Epoch 6/10
632/632 [==============================] - 135s 213ms/step - loss: 0.8629
Epoch 7/10
632/632 [==============================] - 135s 214ms/step - loss: 0.8083
Epoch 8/10
632/632 [==============================] - 132s 208ms/step - loss: 0.7585
Epoch 9/10
632/632 [==============================] - 133s 210ms/step - loss: 0.7113
Epoch 10/10
632/632 [==============================] - 132s 209ms/step - loss: 0.6664
Генерация из посева: im tortoise--’ ‘why did you call him tortoise, if he wasn’t one?’ alice asked. ‘we called him tortoise because he taught
im tortoise--’ ‘why did you call him tortoise, if he wasn’t one?’ alice asked. ‘we called him tortoise because he taught is to be off i say when you can’t hild be for itserse, down here, and the strangh of course, what w==================================================
Итерация #: 2
Epoch 1/10
632/632 [==============================] - 131s 207ms/step - loss: 0.6289
Epoch 2/10
632/632 [==============================] - 134s 211ms/step - loss: 0.5920
Epoch 3/10
632/632 [==============================] - 134s 211ms/step - loss: 0.5589
Epoch 4/10
632/632 [==============================] - 133s 210ms/step - loss: 0.5295
Epoch 5/10
632/632 [==============================] - 136s 215ms/step - loss: 0.5032
Epoch 6/10
632/632 [==============================] - 136s 215ms/step - loss: 0.4774
Epoch 7/10
632/632 [==============================] - 136s 215ms/step - loss: 0.4580
Epoch 8/10
632/632 [==============================] - 134s 213ms/step - loss: 0.4369
Epoch 9/10
632/632 [==============================] - 133s 210ms/step - loss: 0.4189
Epoch 10/10
632/632 [==============================] - 131s 207ms/step - loss: 0.4027
Генерация из посева: oly air, and, after folding his arms and frowning at the cook till his eyes were nearly out of sight, he said in a deep 
oly air, and, after folding his arms and frowning at the cook till his eyes were nearly out of sight, he said in a deep to be the project gutenberg-tm were armplly following ereating the foundation, the ond to ento you c==================================================
Итерация #: 3
Epoch 1/10
632/632 [==============================] - 131s 208ms/step - loss: 0.3883
Epoch 2/10
632/632 [==============================] - 134s 211ms/step - loss: 0.3749
Epoch 3/10
632/632 [==============================] - 133s 210ms/step - loss: 0.3627
Epoch 4/10
632/632 [==============================] - 128s 203ms/step - loss: 0.3502
Epoch 5/10
632/632 [==============================] - 133s 210ms/step - loss: 0.3398
Epoch 6/10
632/632 [==============================] - 133s 210ms/step - loss: 0.3291
Epoch 7/10
632/632 [==============================] - 134s 212ms/step - loss: 0.3215
Epoch 8/10
632/632 [==============================] - 133s 211ms/step - loss: 0.3135
Epoch 9/10
632/632 [==============================] - 134s 211ms/step - loss: 0.3079
Epoch 10/10
632/632 [==============================] - 133s 210ms/step - loss: 0.2990
Генерация из посева: rink any further: she felt a little nervous about this; ‘for it might end, you know,’ said alice to herself, ‘in my goin
rink any further: she felt a little nervous about this; ‘for it might end, you know,’ said alice to herself, ‘in my going of a whise prepent, i should like to dubber the fourthersing of the sort, and the mock turtle soun==================================================
Итерация #: 4
Epoch 1/10
632/632 [==============================] - 133s 211ms/step - loss: 0.2918
Epoch 2/10
632/632 [==============================] - 135s 214ms/step - loss: 0.2831
Epoch 3/10
632/632 [==============================] - 135s 214ms/step - loss: 0.2782
Epoch 4/10
632/632 [==============================] - 135s 213ms/step - loss: 0.2744
Epoch 5/10
632/632 [==============================] - 134s 212ms/step - loss: 0.2684
Epoch 6/10
632/632 [==============================] - 132s 208ms/step - loss: 0.2603
Epoch 7/10
632/632 [==============================] - 133s 210ms/step - loss: 0.2595
Epoch 8/10
632/632 [==============================] - 132s 209ms/step - loss: 0.2523
Epoch 9/10
632/632 [==============================] - 135s 213ms/step - loss: 0.2503
Epoch 10/10
632/632 [==============================] - 135s 213ms/step - loss: 0.2469
Генерация из посева: portant--’ as if he were trying which word sounded best. some of the jury wrote it down ‘important,’ and some ‘unimporta
portant--’ as if he were trying which word sounded best. some of the jury wrote it down ‘important,’ and some ‘unimportant. off com! the rool wele peepenss letter not to be over here, and when the crousides wighad a tria==================================================
Итерация #: 5
Epoch 1/10
632/632 [==============================] - 132s 209ms/step - loss: 0.2396
Epoch 2/10
632/632 [==============================] - 131s 207ms/step - loss: 0.2387
Epoch 3/10
632/632 [==============================] - 134s 212ms/step - loss: 0.2352
Epoch 4/10
632/632 [==============================] - 133s 210ms/step - loss: 0.2299
Epoch 5/10
632/632 [==============================] - 134s 212ms/step - loss: 0.2289
Epoch 6/10
632/632 [==============================] - 135s 214ms/step - loss: 0.2250
Epoch 7/10
632/632 [==============================] - 135s 214ms/step - loss: 0.2202
Epoch 8/10
632/632 [==============================] - 137s 217ms/step - loss: 0.2181
Epoch 9/10
632/632 [==============================] - 136s 216ms/step - loss: 0.2135
Epoch 10/10
632/632 [==============================] - 135s 213ms/step - loss: 0.2114
Генерация из посева: ancient and modern, with seaography: then drawling--the drawling-master was an old conger-eel, that used to come once a 
ancient and modern, with seaography: then drawling--the drawling-master was an old conger-eel, that used to come once a weets!’ ‘she drewe now, me courde, and the mock out to ore offe.’ ‘you din!’ said the caterpillar. a==================================================
Итерация #: 6
Epoch 1/10
632/632 [==============================] - 135s 213ms/step - loss: 0.2126
Epoch 2/10
632/632 [==============================] - 134s 212ms/step - loss: 0.2081
Epoch 3/10
632/632 [==============================] - 134s 212ms/step - loss: 0.2036
Epoch 4/10
632/632 [==============================] - 131s 207ms/step - loss: 0.2053
Epoch 5/10
632/632 [==============================] - 131s 208ms/step - loss: 0.2021
Epoch 6/10
632/632 [==============================] - 133s 211ms/step - loss: 0.2015
Epoch 7/10
632/632 [==============================] - 137s 216ms/step - loss: 0.1981
Epoch 8/10
632/632 [==============================] - 136s 215ms/step - loss: 0.1963
Epoch 9/10
632/632 [==============================] - 136s 216ms/step - loss: 0.1960
Epoch 10/10
632/632 [==============================] - 136s 215ms/step - loss: 0.1952
Генерация из посева:  the hatter. ‘nor i,’ said the march hare. alice sighed wearily. ‘i think you might do something better with the time,’ 
 the hatter. ‘nor i,’ said the march hare. alice sighed wearily. ‘i think you might do something better with the time,’ said the caterpillar. ‘i never was so much!’ said alice, sharp about the tired of the out, and the o==================================================
Итерация #: 7
Epoch 1/10
632/632 [==============================] - 134s 213ms/step - loss: 0.1909
Epoch 2/10
632/632 [==============================] - 135s 213ms/step - loss: 0.1904
Epoch 3/10
632/632 [==============================] - 137s 216ms/step - loss: 0.1884
Epoch 4/10
632/632 [==============================] - 136s 215ms/step - loss: 0.1867
Epoch 5/10
632/632 [==============================] - 137s 217ms/step - loss: 0.1822
Epoch 6/10
632/632 [==============================] - 136s 216ms/step - loss: 0.1810
Epoch 7/10
632/632 [==============================] - 136s 215ms/step - loss: 0.1840
Epoch 8/10
632/632 [==============================] - 136s 215ms/step - loss: 0.1824
Epoch 9/10
632/632 [==============================] - 134s 213ms/step - loss: 0.1805
Epoch 10/10
632/632 [==============================] - 137s 217ms/step - loss: 0.1761
Генерация из посева: lle, that she hardly knew what she was saying, and the words came very queer indeed:-- ‘’tis the voice of the lobster; i
lle, that she hardly knew what she was saying, and the words came very queer indeed:-- ‘’tis the voice of the lobster; i will just be stoppinghey with hish?’ ‘oh,’ said the hatter. ‘i this quite it mist, you never a chis==================================================
Итерация #: 8
Epoch 1/10
632/632 [==============================] - 135s 213ms/step - loss: 0.1759
Epoch 2/10
632/632 [==============================] - 133s 211ms/step - loss: 0.1765
Epoch 3/10
632/632 [==============================] - 133s 211ms/step - loss: 0.1748
Epoch 4/10
632/632 [==============================] - 133s 210ms/step - loss: 0.1715
Epoch 5/10
632/632 [==============================] - 132s 209ms/step - loss: 0.1701
Epoch 6/10
632/632 [==============================] - 131s 208ms/step - loss: 0.1688
Epoch 7/10
632/632 [==============================] - 133s 211ms/step - loss: 0.1696
Epoch 8/10
632/632 [==============================] - 131s 208ms/step - loss: 0.1656
Epoch 9/10
632/632 [==============================] - 132s 209ms/step - loss: 0.1645
Epoch 10/10
632/632 [==============================] - 132s 208ms/step - loss: 0.1656
Генерация из посева: ddy.’ ‘all right,’ said the cat; and this time it vanished quite slowly, beginning with the end of the tail, and ending 
ddy.’ ‘all right,’ said the cat; and this time it vanished quite slowly, beginning with the end of the tail, and ending when she could not tell read, in favigut against carried in her hand, and a little distance was she ==================================================
Итерация #: 9
Epoch 1/10
632/632 [==============================] - 132s 209ms/step - loss: 0.1617
Epoch 2/10
632/632 [==============================] - 132s 209ms/step - loss: 0.1594
Epoch 3/10
632/632 [==============================] - 131s 208ms/step - loss: 0.1589
Epoch 4/10
632/632 [==============================] - 131s 207ms/step - loss: 0.1581
Epoch 5/10
632/632 [==============================] - 132s 208ms/step - loss: 0.1589
Epoch 6/10
632/632 [==============================] - 130s 205ms/step - loss: 0.1613
Epoch 7/10
632/632 [==============================] - 130s 206ms/step - loss: 0.1583
Epoch 8/10
632/632 [==============================] - 130s 205ms/step - loss: 0.1555
Epoch 9/10
632/632 [==============================] - 132s 208ms/step - loss: 0.1589
Epoch 10/10
632/632 [==============================] - 129s 204ms/step - loss: 0.1565
Генерация из посева: his last word two or three times over to herself, being rather proud of it: for she thought, and rightly too, that very 
his last word two or three times over to herself, being rather proud of it: for she thought, and rightly too, that very fol, coryt into that alice heard of sog, ‘i don’t say,’ said the cook. to the lost cooked at the mar

BATCH_SIZE, HIDDEN_SIZE = 64, 256
NUM_ITERATIONS = 2
NUM_EPOCHS_PER_ITERATION = 20
NUM_PREDS_PER_EPOCH = 100



==================================================
Итерация #: 0
Epoch 1/2
1264/1264 [==============================] - 227s 172ms/step - loss: 2.2996
Epoch 2/2
1264/1264 [==============================] - 231s 182ms/step - loss: 1.8962
Генерация из посева: --” how are you getting on now, my dear?’ it continued, turning to alice as it spoke. ‘as wet as ever,’ said alice in a 
--” how are you getting on now, my dear?’ it continued, turning to alice as it spoke. ‘as wet as ever,’ said alice in a little said the had so her her her her her her her her her her her her her her her her her her her h==================================================
Итерация #: 1
Epoch 1/2
1264/1264 [==============================] - 234s 185ms/step - loss: 1.7151
Epoch 2/2
1264/1264 [==============================] - 233s 184ms/step - loss: 1.5840
Генерация из посева: ase your majesty,’ said alice very politely; but she added, to herself, ‘why, they’re only a pack of cards, after all. i
ase your majesty,’ said alice very politely; but she added, to herself, ‘why, they’re only a pack of cards, after all. it was a little the rabbit said to her hear she hed done of the rabbit said to her hear she hed done ==================================================
Итерация #: 2
Epoch 1/2
1264/1264 [==============================] - 234s 185ms/step - loss: 1.4783
Epoch 2/2
1264/1264 [==============================] - 230s 182ms/step - loss: 1.3823
Генерация из посева: side,’ the queen added to one of the officers: but the hatter was out of sight before the officer could get to the door.
side,’ the queen added to one of the officers: but the hatter was out of sight before the officer could get to the door.  here was very suld the stopent of the stopent of the stopent of the stopent of the stopent of the ==================================================
Итерация #: 3
Epoch 1/2
1264/1264 [==============================] - 235s 186ms/step - loss: 1.2954
Epoch 2/2
1264/1264 [==============================] - 236s 187ms/step - loss: 1.2171
Генерация из посева:  me see--how is it to be managed? i suppose i ought to eat or drink something or other; but the great question is, what?
 me see--how is it to be managed? i suppose i ought to eat or drink something or other; but the great question is, what?’ said the duchess seeped to herself she had not and the project gutenberg-tm electronic works and t==================================================
Итерация #: 4
Epoch 1/2
1264/1264 [==============================] - 240s 190ms/step - loss: 1.1397
Epoch 2/2
1264/1264 [==============================] - 230s 182ms/step - loss: 1.0668
Генерация из посева: r, if you don’t like them!’ when the mouse heard this, it turned round and swam slowly back to her: its face was quite p
r, if you don’t like them!’ when the mouse heard this, it turned round and swam slowly back to her: its face was quite patter of the mouse, and she thought in a great hunder which was she went on encoming of the gryphon,==================================================
Итерация #: 5
Epoch 1/2
1264/1264 [==============================] - 229s 181ms/step - loss: 0.9961
Epoch 2/2
1264/1264 [==============================] - 230s 182ms/step - loss: 0.9309
Генерация из посева: ctly idiotic!’ and she opened the door and went in. the door led right into a large kitchen, which was full of smoke fro
ctly idiotic!’ and she opened the door and went in. the door led right into a large kitchen, which was full of smoke from other the singeration of the other, and the project gutenberg-tm license of the project gutenberg-==================================================
Итерация #: 6
Epoch 1/2
1264/1264 [==============================] - 233s 184ms/step - loss: 0.8706
Epoch 2/2
1264/1264 [==============================] - 236s 187ms/step - loss: 0.8148
Генерация из посева: ock turtle a little anxiously. ‘yes,’ said alice, ‘we learned french and music.’ ‘and washing?’ said the mock turtle. ‘c
ock turtle a little anxiously. ‘yes,’ said alice, ‘we learned french and music.’ ‘and washing?’ said the mock turtle. ‘chapge in at the sears of much past the eremory of the seary of curtiess creeter, such as it doenngru==================================================
Итерация #: 7
Epoch 1/2
1264/1264 [==============================] - 235s 186ms/step - loss: 0.7668
Epoch 2/2
1264/1264 [==============================] - 236s 187ms/step - loss: 0.7234
Генерация из посева: bout two feet high: even then she walked up towards it rather timidly, saying to herself ‘suppose it should be raving ma
bout two feet high: even then she walked up towards it rather timidly, saying to herself ‘suppose it should be raving mad--o he promectute peeser, you know.’ ‘a don’t believe she had no more then the mock turtle was noth==================================================
Итерация #: 8
Epoch 1/2
1264/1264 [==============================] - 233s 185ms/step - loss: 0.6838
Epoch 2/2
1264/1264 [==============================] - 235s 186ms/step - loss: 0.6498
Генерация из посева: son, ‘i feared it might injure the brain; but, now that i’m perfectly sure i have none, why, i do it again and again.’ ‘
son, ‘i feared it might injure the brain; but, now that i’m perfectly sure i have none, why, i do it again and again.’ ‘you are orded ig is my dian you may never every now are you think you can do you think you cone to t==================================================
Итерация #: 9
Epoch 1/2
1264/1264 [==============================] - 234s 185ms/step - loss: 0.6227
Epoch 2/2
1264/1264 [==============================] - 229s 181ms/step - loss: 0.5945
Генерация из посева: he king. ‘it began with the tea,’ the hatter replied. ‘of course twinkling begins with a t!’ said the king sharply. ‘do 
he king. ‘it began with the tea,’ the hatter replied. ‘of course twinkling begins with a t!’ said the king sharply. ‘do you thinking on the baby moself, as she was see it say, she went remember in that is the sead of the==================================================
Итерация #: 10
Epoch 1/2
1264/1264 [==============================] - 229s 181ms/step - loss: 0.5724
Epoch 2/2
1264/1264 [==============================] - 234s 185ms/step - loss: 0.5543
Генерация из посева:  her so savage when they met in the kitchen. ‘when i’m a duchess,’ she said to herself, (not in a very hopeful tone thou
 her so savage when they met in the kitchen. ‘when i’m a duchess,’ she said to herself, (not in a very hopeful tone though. what fanned how sole the stook in the middle than at it: the mock turtle waitent to the table, b==================================================
Итерация #: 11
Epoch 1/2
1264/1264 [==============================] - 232s 184ms/step - loss: 0.5341
Epoch 2/2
1264/1264 [==============================] - 233s 184ms/step - loss: 0.5227
Генерация из посева: d to ask his neighbour to tell him. ‘a nice muddle their slates’ll be in before the trial’s over!’ thought alice. one of
d to ask his neighbour to tell him. ‘a nice muddle their slates’ll be in before the trial’s over!’ thought alice. one of the ground to see its mase a copitsand without a great courater the shode, ‘i don’t see as well as ==================================================
Итерация #: 12
Epoch 1/2
1264/1264 [==============================] - 234s 185ms/step - loss: 0.5102
Epoch 2/2
1264/1264 [==============================] - 239s 189ms/step - loss: 0.4964
Генерация из посева: y for alice, the little magic bottle had now had its full effect, and she grew no larger: still it was very uncomfortabl
y for alice, the little magic bottle had now had its full effect, and she grew no larger: still it was very uncomfortable. the mock turtle said, sure they lef she tried to be the shild have when it said voice. ‘i upon it==================================================
Итерация #: 13
Epoch 1/2
1264/1264 [==============================] - 239s 189ms/step - loss: 0.4856
Epoch 2/2
1264/1264 [==============================] - 239s 189ms/step - loss: 0.4713
Генерация из посева: ing is, to get into that beautiful garden--how is that to be done, i wonder?’ as she said this, she came suddenly upon a
ing is, to get into that beautiful garden--how is that to be done, i wonder?’ as she said this, she came suddenly upon anything that soundering and project gutenberg-tm electronic works, and the fance with the project gu==================================================
Итерация #: 14
Epoch 1/2
1264/1264 [==============================] - 238s 188ms/step - loss: 0.4626
Epoch 2/2
1264/1264 [==============================] - 237s 188ms/step - loss: 0.4536
Генерация из посева: still it was very uncomfortable, and, as there seemed to be no sort of chance of her ever getting out of the room again,
still it was very uncomfortable, and, as there seemed to be no sort of chance of her ever getting out of the room again, and she went dew very again, with a courat of her asss, there was a selpeally het had a little not,==================================================
Итерация #: 15
Epoch 1/2
1264/1264 [==============================] - 236s 187ms/step - loss: 0.4500
Epoch 2/2
1264/1264 [==============================] - 238s 188ms/step - loss: 0.4410
Генерация из посева:  witness would be like, ‘--for they haven’t got much evidence yet,’ she said to herself. imagine her surprise, when the 
 witness would be like, ‘--for they haven’t got much evidence yet,’ she said to herself. imagine her surprise, when the tide wish the wonder as it was in the pool again the mock turtle your wook the great queer without t==================================================
Итерация #: 16
Epoch 1/2
1264/1264 [==============================] - 237s 188ms/step - loss: 0.4315
Epoch 2/2
1264/1264 [==============================] - 234s 186ms/step - loss: 0.4244
Генерация из посева: er: its face was quite pale (with passion, alice thought), and it said in a low trembling voice, ‘let us get to the shor
er: its face was quite pale (with passion, alice thought), and it said in a low trembling voice, ‘let us get to the shore it looking ala mouse--and yes, she though of mich,’ said alice. ‘what way that i whow what was ’ u==================================================
Итерация #: 17
Epoch 1/2
1264/1264 [==============================] - 233s 184ms/step - loss: 0.4193
Epoch 2/2
1264/1264 [==============================] - 236s 187ms/step - loss: 0.4124
Генерация из посева:  better with the time,’ she said, ‘than waste it in asking riddles that have no answers.’ ‘if you knew time as well as i
 better with the time,’ she said, ‘than waste it in asking riddles that have no answers.’ ‘if you knew time as well as i don’t know that ith in by till it’s the first paster wrattyring, perhaps not chee it seems to say w==================================================
Итерация #: 18
Epoch 1/2
1264/1264 [==============================] - 231s 183ms/step - loss: 0.4088
Epoch 2/2
1264/1264 [==============================] - 233s 184ms/step - loss: 0.4069
Генерация из посева: asted the rest of my life.’ ‘you are old,’ said the youth, ‘one would hardly suppose that your eye was as steady as ever
asted the rest of my life.’ ‘you are old,’ said the youth, ‘one would hardly suppose that your eye was as steady as everything that it make a one elless is it a without what you walking of the works els are molentely the==================================================
Итерация #: 19
Epoch 1/2
1264/1264 [==============================] - 234s 185ms/step - loss: 0.4018
Epoch 2/2
1264/1264 [==============================] - 234s 185ms/step - loss: 0.3952
Генерация из посева: shop of canterbury, found it advisable--“’ ‘found what?’ said the duck. ‘found it,’ the mouse replied rather crossly: ‘o
shop of canterbury, found it advisable--“’ ‘found what?’ said the duck. ‘found it,’ the mouse replied rather crossly: ‘of course not think you’re then, you know,’ the mock turtle was not as of the room, with the other, t==================================================
Итерация #: 20
Epoch 1/2
1264/1264 [==============================] - 231s 183ms/step - loss: 0.3912
Epoch 2/2
1264/1264 [==============================] - 235s 186ms/step - loss: 0.3882
Генерация из посева:  nice muddle their slates’ll be in before the trial’s over!’ thought alice. one of the jurors had a pencil that squeaked
 nice muddle their slates’ll be in before the trial’s over!’ thought alice. one of the jurors had a pencil that squeaked. this agreement.  for to the singerly, and alice was so alice’s donit vent get s: mund the lobster ==================================================
Итерация #: 21
Epoch 1/2
1264/1264 [==============================] - 238s 188ms/step - loss: 0.3829
Epoch 2/2
1264/1264 [==============================] - 239s 189ms/step - loss: 0.3775
Генерация из посева: l it her,’ said the mock turtle in a deep, hollow tone: ‘sit down, both of you, and don’t speak a word till i’ve finishe
l it her,’ said the mock turtle in a deep, hollow tone: ‘sit down, both of you, and don’t speak a word till i’ve finished it till she was out of sigh, and the brightered about the sindering would not remark right had a l==================================================
Итерация #: 22
Epoch 1/2
1264/1264 [==============================] - 245s 194ms/step - loss: 0.3769
Epoch 2/2
1264/1264 [==============================] - 244s 193ms/step - loss: 0.3736
Генерация из посева: e cat. ‘i said pig,’ replied alice; ‘and i wish you wouldn’t keep appearing and vanishing so suddenly: you make one quit
e cat. ‘i said pig,’ replied alice; ‘and i wish you wouldn’t keep appearing and vanishing so suddenly: you make one quite got better,’ the mock turtle went on, ‘a dvey nothing ensellons copyright it it might hemp the att==================================================
Итерация #: 23
Epoch 1/2
1264/1264 [==============================] - 239s 189ms/step - loss: 0.3650
Epoch 2/2
1264/1264 [==============================] - 240s 190ms/step - loss: 0.3599
Генерация из посева:  alice for protection. ‘you shan’t be beheaded!’ said alice, and she put them into a large flower-pot that stood near. t
 alice for protection. ‘you shan’t be beheaded!’ said alice, and she put them into a large flower-pot that stood near. the fart falis in the dast of a this wook of reading project gutenberg-tm electronic work in a fistan==================================================
Итерация #: 24
Epoch 1/2
1264/1264 [==============================] - 244s 193ms/step - loss: 0.3581
Epoch 2/2
1264/1264 [==============================] - 244s 193ms/step - loss: 0.3591
Генерация из посева: d we put a white one in by mistake; and if the queen was to find it out, we should all have our heads cut off, you know.
d we put a white one in by mistake; and if the queen was to find it out, we should all have our heads cut off, you know. ‘it was a curious derestont, fallen them,’ shouted the queen, ‘i can’t remember her tainsd,’ said a

BATCH_SIZE, HIDDEN_SIZE = 64, 256
NUM_ITERATIONS = 25
NUM_EPOCHS_PER_ITERATION = 20
NUM_PREDS_PER_EPOCH = 100

==================================================
Итерация #: 0
Epoch 1/20
1264/1264 [==============================] - 556s 411ms/step - loss: 2.3636
Epoch 2/20
1264/1264 [==============================] - 517s 409ms/step - loss: 1.9781
Epoch 3/20
1264/1264 [==============================] - 513s 406ms/step - loss: 1.7984
Epoch 4/20
1264/1264 [==============================] - 523s 414ms/step - loss: 1.6604
Epoch 5/20
1264/1264 [==============================] - 501s 396ms/step - loss: 1.5483
Epoch 6/20
1264/1264 [==============================] - 503s 398ms/step - loss: 1.4532
Epoch 7/20
1264/1264 [==============================] - 499s 395ms/step - loss: 1.3692
Epoch 8/20
1264/1264 [==============================] - 495s 391ms/step - loss: 1.2932
Epoch 9/20
1264/1264 [==============================] - 498s 394ms/step - loss: 1.2186
Epoch 10/20
1264/1264 [==============================] - 498s 394ms/step - loss: 1.1486
Epoch 11/20
1264/1264 [==============================] - 499s 395ms/step - loss: 1.0786
Epoch 12/20
1264/1264 [==============================] - 502s 397ms/step - loss: 1.0123
Epoch 13/20
1264/1264 [==============================] - 493s 390ms/step - loss: 0.9476
Epoch 14/20
1264/1264 [==============================] - 486s 384ms/step - loss: 0.8853
Epoch 15/20
1264/1264 [==============================] - 490s 387ms/step - loss: 0.8306
Epoch 16/20
1264/1264 [==============================] - 487s 385ms/step - loss: 0.7792
Epoch 17/20
1264/1264 [==============================] - 491s 389ms/step - loss: 0.7340
Epoch 18/20
1264/1264 [==============================] - 488s 386ms/step - loss: 0.6957
Epoch 19/20
1264/1264 [==============================] - 490s 388ms/step - loss: 0.6606
Epoch 20/20
1264/1264 [==============================] - 488s 386ms/step - loss: 0.6324
Генерация из посева: ment.  there are a few things that you can do with most project gutenberg-tm electronic works even without complying wit
ment.  there are a few things that you can do with most project gutenberg-tm electronic works even without complying with the little door said to herself, ‘that’s very coriate fread all the terms of this cardered again. ==================================================
Итерация #: 1
Epoch 1/20
1264/1264 [==============================] - 515s 407ms/step - loss: 0.6043
Epoch 2/20
1264/1264 [==============================] - 509s 402ms/step - loss: 0.5836
Epoch 3/20
1264/1264 [==============================] - 515s 407ms/step - loss: 0.5654
Epoch 4/20
1264/1264 [==============================] - 527s 417ms/step - loss: 0.5481
Epoch 5/20
1264/1264 [==============================] - 518s 410ms/step - loss: 0.5357
Epoch 6/20
1264/1264 [==============================] - 510s 404ms/step - loss: 0.5205
Epoch 7/20
1264/1264 [==============================] - 515s 408ms/step - loss: 0.5112
Epoch 8/20
1264/1264 [==============================] - 518s 410ms/step - loss: 0.5013
Epoch 9/20
1264/1264 [==============================] - 530s 419ms/step - loss: 0.4885
Epoch 10/20
1264/1264 [==============================] - 514s 407ms/step - loss: 0.4813
Epoch 11/20
1264/1264 [==============================] - 499s 395ms/step - loss: 0.4764
Epoch 12/20
1264/1264 [==============================] - 495s 392ms/step - loss: 0.4676
Epoch 13/20
1264/1264 [==============================] - 495s 391ms/step - loss: 0.4603
Epoch 14/20
1264/1264 [==============================] - 506s 401ms/step - loss: 0.4525
Epoch 15/20
1264/1264 [==============================] - 502s 397ms/step - loss: 0.4491
Epoch 16/20
1264/1264 [==============================] - 496s 393ms/step - loss: 0.4411
Epoch 17/20
1264/1264 [==============================] - 497s 393ms/step - loss: 0.4369
Epoch 18/20
1264/1264 [==============================] - 491s 388ms/step - loss: 0.4321
Epoch 19/20
1264/1264 [==============================] - 499s 395ms/step - loss: 0.4291
Epoch 20/20
1264/1264 [==============================] - 499s 395ms/step - loss: 0.4252
Генерация из посева: e rattle of the teacups as the march hare and his friends shared their never-ending meal, and the shrill voice of the qu
e rattle of the teacups as the march hare and his friends shared their never-ending meal, and the shrill voice of the queen of this agreement for see hurried off a geesing its hutter. ‘any would go round and growing to h==================================================
Итерация #: 2
Epoch 1/20
1264/1264 [==============================] - 503s 398ms/step - loss: 0.4205
Epoch 2/20
1264/1264 [==============================] - 503s 398ms/step - loss: 0.4179
Epoch 3/20
1264/1264 [==============================] - 504s 399ms/step - loss: 0.4107
Epoch 4/20
1264/1264 [==============================] - 504s 399ms/step - loss: 0.4105
Epoch 5/20
1264/1264 [==============================] - 506s 400ms/step - loss: 0.4063
Epoch 6/20
1264/1264 [==============================] - 507s 401ms/step - loss: 0.4031
Epoch 7/20
1264/1264 [==============================] - 510s 404ms/step - loss: 0.3948
Epoch 8/20
1264/1264 [==============================] - 508s 402ms/step - loss: 0.3970
Epoch 9/20
1264/1264 [==============================] - 508s 402ms/step - loss: 0.3924
Epoch 10/20
1264/1264 [==============================] - 509s 402ms/step - loss: 0.3898
Epoch 11/20
1264/1264 [==============================] - 509s 403ms/step - loss: 0.3862
Epoch 12/20
1264/1264 [==============================] - 511s 404ms/step - loss: 0.3862
Epoch 13/20
1264/1264 [==============================] - 512s 405ms/step - loss: 0.3794
Epoch 14/20
1264/1264 [==============================] - 509s 403ms/step - loss: 0.3797
Epoch 15/20
1264/1264 [==============================] - 509s 403ms/step - loss: 0.3720
Epoch 16/20
1264/1264 [==============================] - 507s 401ms/step - loss: 0.3703
Epoch 17/20
1264/1264 [==============================] - 504s 399ms/step - loss: 0.3739
Epoch 18/20
1264/1264 [==============================] - 502s 397ms/step - loss: 0.3717
Epoch 19/20
1264/1264 [==============================] - 501s 396ms/step - loss: 0.3670
Epoch 20/20
1264/1264 [==============================] - 494s 391ms/step - loss: 0.3664
Генерация из посева: u know about this business?’ the king said to alice. ‘nothing,’ said alice. ‘nothing whatever?’ persisted the king. ‘not
u know about this business?’ the king said to alice. ‘nothing,’ said alice. ‘nothing whatever?’ persisted the king. ‘nothing more to do right, you know. they looked at alice’s are lang a distering off all said, and then ==================================================
Итерация #: 3
Epoch 1/20
1264/1264 [==============================] - 506s 400ms/step - loss: 0.3673
Epoch 2/20
1264/1264 [==============================] - 505s 400ms/step - loss: 0.3634
Epoch 3/20
1264/1264 [==============================] - 507s 401ms/step - loss: 0.3623
Epoch 4/20
1264/1264 [==============================] - 503s 398ms/step - loss: 0.3598
Epoch 5/20
1264/1264 [==============================] - 488s 386ms/step - loss: 0.3626
Epoch 6/20
1264/1264 [==============================] - 467s 370ms/step - loss: 0.3561
Epoch 7/20
1264/1264 [==============================] - 468s 370ms/step - loss: 0.3493
Epoch 8/20
1264/1264 [==============================] - 481s 381ms/step - loss: 0.3496
Epoch 9/20
1264/1264 [==============================] - 483s 382ms/step - loss: 0.3502
Epoch 10/20
1264/1264 [==============================] - 485s 384ms/step - loss: 0.3434
Epoch 11/20
1264/1264 [==============================] - 479s 379ms/step - loss: 0.3461
Epoch 12/20
1264/1264 [==============================] - 476s 377ms/step - loss: 0.3419
Epoch 13/20
1264/1264 [==============================] - 479s 379ms/step - loss: 0.3438
Epoch 14/20
1264/1264 [==============================] - 477s 377ms/step - loss: 0.3410
Epoch 15/20
1264/1264 [==============================] - 478s 378ms/step - loss: 0.3368
Epoch 16/20
1264/1264 [==============================] - 481s 381ms/step - loss: 0.3390
Epoch 17/20
1264/1264 [==============================] - 480s 380ms/step - loss: 0.3422
Epoch 18/20
1264/1264 [==============================] - 483s 382ms/step - loss: 0.3370
Epoch 19/20
1264/1264 [==============================] - 487s 385ms/step - loss: 0.3356
Epoch 20/20
1264/1264 [==============================] - 475s 376ms/step - loss: 0.3337
Генерация из посева: h 1.f.3, the project gutenberg literary archive foundation, the owner of the project gutenberg-tm trademark, and any oth
h 1.f.3, the project gutenberg literary archive foundation, the owner of the project gutenberg-tm trademark, and any other little thing was a parrance’s any other was a longer its mock turtle so, foromy, though saye this==================================================
Итерация #: 4
Epoch 1/20
1264/1264 [==============================] - 492s 389ms/step - loss: 0.3294
Epoch 2/20
1264/1264 [==============================] - 490s 388ms/step - loss: 0.3315
Epoch 3/20
1264/1264 [==============================] - 478s 378ms/step - loss: 0.3268
Epoch 4/20
1264/1264 [==============================] - 495s 392ms/step - loss: 0.3270
Epoch 5/20
1264/1264 [==============================] - 489s 387ms/step - loss: 0.3235
Epoch 6/20
 386/1264 [========>.....................] - ETA: 5:36 - loss: 0.2754